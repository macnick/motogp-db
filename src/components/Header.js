import React from 'react';
import logo from '../assets/img/motogp.png';

const Header = () => (
  <div className="header">
    <div className="logo">
      <a href="/">
        <img src={logo} alt="MotoGP Logo" height="120px" width="120px" />
        <span className="info">MotoGP Team Info</span>
      </a>
    </div>
  </div>
);

export default Header;
