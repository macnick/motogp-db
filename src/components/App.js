import React from 'react';
import Header from './Header';
import TeamList from '../containers/TeamList';

const App = () => (
  <div className="app">
    <Header />
    <TeamList />
  </div>
);

export default App;
