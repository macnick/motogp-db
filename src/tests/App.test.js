import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import App from '../components/App';
import Header from '../components/Header';
import '@testing-library/jest-dom';

const mockStore = configureStore([]);

describe('Renders my Redux connected component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      teamsList: {
        loading: false,
        teams: [
          {
            idTeam: '137292',
            strTeamBadge:
              'https://www.thesportsdb.com/images/media/team/badge/qvsw871531399264.png',
            strTeam: 'Repsol Honda',
            strCountry: 'Japan',
          },
        ],
        team: {
          strTeam: '',
        },
      },
      filter: 'All',
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
  });

  afterEach(cleanup);

  it('should render the App', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the Header', () => {
    const { getByText } = render(<Header />);
    expect(getByText(/MotoGP Team Info/)).toBeInTheDocument();
  });

  it('should render the selection menu', () => {
    const { getByText } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(getByText(/Select Country/)).toBeInTheDocument();
  });

  it('should change selection to Italy', () => {
    const { getByTestId, getByRole } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    fireEvent.select(getByRole('combobox'), { target: { value: 'Italy' } });
    expect(getByTestId('head-select')).toHaveValue('Italy');
  });
});
