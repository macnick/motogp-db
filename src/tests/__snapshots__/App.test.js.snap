// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders my Redux connected component should render the App 1`] = `
<div
  className="app"
>
  <div
    className="header"
  >
    <div
      className="logo"
    >
      <a
        href="/"
      >
        <img
          alt="MotoGP Logo"
          height="120px"
          src="motogp.png"
          width="120px"
        />
        <span
          className="info"
        >
          MotoGP Team Info
        </span>
      </a>
    </div>
  </div>
  <div
    id="select"
  >
    Select Country
     
    <select
      data-testid="head-select"
      name="category"
      onChange={[Function]}
      value="All"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Austria"
      >
        Austria
      </option>
      <option
        value="France"
      >
        France
      </option>
      <option
        value="Italy"
      >
        Italy
      </option>
      <option
        value="Japan"
      >
        Japan
      </option>
      <option
        value="Malaysia"
      >
        Malaysia
      </option>
      <option
        value="Spain"
      >
        Spain
      </option>
    </select>
  </div>
  <div
    className="container"
  >
    <a
      href="/team/137292"
      onClick={[Function]}
    >
      <div
        className="line"
        id="137292"
      >
        <div
          className="badge"
        >
          <img
            alt="Repsol Honda team badge"
            src="https://www.thesportsdb.com/images/media/team/badge/qvsw871531399264.png"
          />
        </div>
      </div>
    </a>
  </div>
</div>
`;
